packages/stark-ui/src/modules/date-range-picker/components/date-range-picker.component.ts

```diff
@@ -24,91 +38,151 @@ const componentName = "stark-date-range-picker";
 	// We need to use host instead of @HostBinding: https://github.com/NationalBankBelgium/stark/issues/664
 	host: {
 		class: componentName
-	}
+	},
+	providers: [
+		{
+			provide: NG_VALUE_ACCESSOR,
+			multi: true,
+			useExisting: StarkDateRangePickerComponent
+		}
+	]
 })
-export class StarkDateRangePickerComponent extends AbstractStarkUiComponent implements OnInit {
+export class StarkDateRangePickerComponent extends AbstractStarkUiComponent implements ControlValueAccessor, OnInit, OnDestroy {
 	/**
-	 * Filter function or a string
-	 * Will be applied to both date-picker
+	 * ControlValueAccessor listener
+	 * @ignore
 	 */
-	@Input()
-	public dateFilter?: StarkDatePickerFilter;
+	private _onTouched: () => void = noop;
 
 	/**
-	 * Timestamp Mask Configuration to apply on the start/end date-picker.
-	 * If `true` is passed, the default mask config is applied: {DEFAULT_DATE_MASK_CONFIG|DEFAULT_DATE_MASK_CONFIG}
-	 * If `false` is passed or if `dateMask` is not present, the directive is disabled.
-	 * If a `StarkTimestampMaskConfig` is passed, it is set as the date mask config.
+	 * ControlValueAccessor listener
+	 * @ignore
 	 */
-	@Input()
-	public dateMask?: StarkDatePickerMaskConfig;
+	private _onChange: (_dateRange: StarkDateRangePickerEvent) => void = noop;
 
 	/**
-	 * Source Date to be bound to the end datepicker model
+	 * Subscriptions to be removed at end of component lifecycle
+	 * @ignore
 	 */
-	@Input()
-	public endDate?: Date;
+	private subs: Subscription[] = [];
 
+	/*--- VIEW CHILDREN ---*/
 	/**
-	 * Label to be displayed in the end datepicker
+	 * Reference to the start datepicker embedded in this component
 	 */
-	@Input()
-	public endDateLabel = "STARK.DATE_RANGE_PICKER.TO";
+	@ViewChild("startPicker")
+	public startPicker!: StarkDatePickerComponent;
 
 	/**
-	 * Maximum date of the end date picker
+	 * Reference to the end datepicker embedded in this component
 	 */
-	@Input()
-	public endMaxDate?: Date;
+	@ViewChild("endPicker")
+	public endPicker!: StarkDatePickerComponent;
+
+	/*--- START-DATE CONFIGURATIONS ---*/
+	private _startBeforeEndVal: ValidatorFn = ({ value }) => {
```

Comment:
Can you rename this to `_startBeforeEndValidator` to make it clear it is a validator?